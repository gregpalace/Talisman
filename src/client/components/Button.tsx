import React from 'react';

import '../../stylesheets/button.scss';

const Button = ({
  icon,
  title,
  iconStyle,
  style,
  indicator,
  onClick,
  small,
}: any) => {
  const handleClick = () => {
    if (typeof onClick === 'function') {
      onClick();
    }
  };

  if (!title) return null;

  return (
    <button
      className={`button__container white-border ${
        small ? 'button__container--small' : ''
      }`}
      style={style || {}}
      onClick={handleClick}
    >
      {icon && (
        <i className={`fa fa-${icon} button__icon`} style={iconStyle || {}}>
          {indicator && <div className='button__indicator' />}
        </i>
      )}

      <span>{title}</span>
    </button>
  );
};

export default Button;
