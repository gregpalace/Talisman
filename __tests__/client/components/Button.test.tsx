import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Button from '../../../src/client/components/Button';

describe('React unit tests', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<Button />);
  });

  describe('<Button />', () => {
    it('should render correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render a button with icon and span tags', () => {
      expect(wrapper.find('button')).toBeDefined();
      expect(wrapper.find('button').prop('className')).toEqual(
        'button__container white-border'
      );
    });
  });
});
