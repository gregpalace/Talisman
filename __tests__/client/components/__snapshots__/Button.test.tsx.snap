// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests <Button /> should render correctly 1`] = `""`;
